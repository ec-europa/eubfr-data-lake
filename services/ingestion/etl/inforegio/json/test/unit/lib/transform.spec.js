import moment from 'moment-timezone';
import mapper from '../../../src/lib/transform';
import testRecord from '../../stubs/record.json';

describe(`Inforegio JSON transformer`, () => {
  moment.tz.setDefault('EST');

  let result = {};

  beforeAll(() => {
    result = mapper(testRecord);
  });

  test(`Throws an error on empty input`, () => {
    expect(mapper).toThrow();
  });

  test(`Produces correct JSON output structure`, () => {
    expect(result).toMatchSnapshot();
  });
});
