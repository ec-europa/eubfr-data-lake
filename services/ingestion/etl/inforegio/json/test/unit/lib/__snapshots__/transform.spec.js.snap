// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DG INFOREGIO JSON transformer Produces correct JSON output structure 1`] = `
Object {
  "action": "",
  "budget": Object {
    "eu_contrib": Object {
      "currency": "",
      "raw": "EUR 909 553",
      "value": 909553,
    },
    "funding_area": Array [
      "European Regional Development Fund",
    ],
    "mmf_heading": "",
    "other_contrib": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "private_fund": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "public_fund": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "total_cost": Object {
      "currency": "",
      "raw": "EUR 1 259 101",
      "value": 1259101,
    },
  },
  "call_year": "",
  "description": "<p>An extension to a boat docked on the river Danube has proved a hugely popular entertainment and cultural hub. A38, a former Ukrainian freighter ship, was initially transformed into a multi-functional space in 2003. With ERDF-funding, an adjacent second unit was constructed in 2010 and has been a resounding success.</p>",
  "ec_priorities": Array [],
  "media": Object {
    "cover_image": "",
    "video": "",
  },
  "period": "2007-2013",
  "programme_name": "",
  "project_id": "3479",
  "project_locations": Array [
    Object {
      "address": "",
      "centroid": null,
      "country_code": "HU",
      "location": null,
      "nuts": Array [
        Object {
          "code": "HU10",
          "level": null,
          "name": "",
          "year": null,
        },
      ],
      "postal_code": "",
      "region": "Közép-Magyarország",
      "town": "",
    },
  ],
  "project_website": "http://ec.europa.eu/regional_policy/en/projects/hungary/renovated-ship-a-unique-venue-on-the-danube",
  "related_links": Array [],
  "results": Object {
    "available": "",
    "result": "",
  },
  "status": "",
  "sub_programme_name": "",
  "success_story": "",
  "themes": Array [
    "Business Support",
    "Tourism and Culture",
    "Urban development",
  ],
  "third_parties": Array [
    Object {
      "address": "Petőfi bridge, Budapest",
      "country": "HU",
      "email": "",
      "name": "A38 Ship Budapest",
      "phone": "",
      "region": "",
      "role": "beneficiary",
      "type": "",
      "website": "",
    },
  ],
  "timeframe": Object {
    "from": "2009-01-01T00:00:00.000Z",
    "to": "2010-11-30T00:00:00.000Z",
  },
  "title": "Renovated ship: a unique venue on the Danube",
  "type": Array [
    "Standard",
  ],
}
`;
